### DISCUSSION TITLE
Why does useRoute return incorrect route?

### DISCUSSION THREAD

**Discussion by MRezaKarimi** [vuejs/router#2041] [Help and Questions] [answered]
Hello everyone.
To understand my question, please consider [this example](https://codesandbox.io/s/quizzical-paper-ds8jhs?file=/src/App.vue). When I click on about link and then refresh the page, `route.path` printed in console is "/" (instead of "/about"). why does this happen and what's the solution?

**Comment by posva** [comment, marked as answer]
Because all navigations are async. The first time the route is ready is after `await router.isReady()`
